src/legacy/core_plugins/console/np_ready/public/application/containers/editor/legacy/console_editor/editor.test.tsx

```diff
@@ -0,0 +1,134 @@
+ * under the License.
+ */
+
+import React from 'react';
+import { mount } from 'enzyme';
+import * as sinon from 'sinon';
+import { EditorContextProvider } from '../../context';
+import { AppContextProvider } from '../../../../context';
+
+import { Editor } from './editor';
+
+jest.mock('../../../../components/editor_example.tsx', () => {});
+jest.mock('../../../../../../../public/quarantined/src/mappings.js', () => ({
+  retrieveAutoCompleteInfo: () => {},
+}));
+jest.mock('../../../../../../../public/quarantined/src/input.js', () => {
+  return {
+    initializeInput: () => ({
+      $el: {
+        css: () => {},
+      },
+      focus: () => {},
+      update: () => {},
+      getSession: () => ({ on: () => {}, setUseWrapMode: () => {} }),
+      commands: {
+        addCommand: () => {},
+      },
+    }),
+  };
+});
+
+import * as sendRequestModule from './send_current_request_to_es';
+import * as consoleMenuActions from '../console_menu_actions';
+
+describe('Legacy (Ace) Console Editor Component Smoke Test', () => {
+  let mockedAppContextValue: any;
+
+  beforeEach(() => {
+    document.queryCommandSupported = sinon.fake(() => true);
+    mockedAppContextValue = {
+      services: {
+        history: {
+          getSavedEditorState: () => null,
+          updateCurrentState: () => {},
+        },
+      },
+      // eslint-disable-next-line
+      ResizeChecker: function() {
+        return { on: () => {} };
+      },
+      docLinkVersion: 'NA',
+    };
+  });
+
+  it('calls send current request to ES', () => {
+    const stub = sinon.stub(sendRequestModule, 'sendCurrentRequestToES');
+    try {
+      const wrapped = mount(
+        <AppContextProvider value={mockedAppContextValue}>
+          <EditorContextProvider settings={{} as any}>
+            <Editor />
+          </EditorContextProvider>
+        </AppContextProvider>
+      );
+      wrapped.find('[data-test-subj~="send-request-button"]').simulate('click');
+      expect(stub.called).toBe(true);
+      expect(stub.callCount).toBe(1);
+    } finally {
+      stub.restore();
+    }
+  });
+
+  it('opens docs', () => {
+    const stub = sinon.stub(consoleMenuActions, 'getDocumentation');
+    try {
+      const wrapped = mount(
+        <AppContextProvider value={mockedAppContextValue}>
+          <EditorContextProvider settings={{} as any}>
+            <Editor />
+          </EditorContextProvider>
+        </AppContextProvider>
+      );
+      const consoleMenuToggle = wrapped.find('[data-test-subj~="toggleConsoleMenu"]').last();
+      consoleMenuToggle.simulate('click');
+
+      const docsButton = wrapped.find('[data-test-subj~="consoleMenuOpenDocs"]').last();
+      docsButton.simulate('click');
+
+      expect(stub.called).toBe(true);
+      expect(stub.callCount).toBe(1);
+    } finally {
+      stub.restore();
+    }
+  });
+
+  it('prompts auto-indent', () => {
+    const stub = sinon.stub(consoleMenuActions, 'autoIndent');
+    try {
+      const wrapped = mount(
```

Comment:
is it necessary to mount the component for each `it()` statement? looks like it is the same across tests.

Reply:
Nope, I guess I was still feeling out what the setup would be for each case. Moved this mounting into the beforeEach.